Lumina

woutervdub opened the pull request:
New design

BenJamesBen reviewed (changes requested):
I tested the PR in the lab.

The Lumina design does _not_ show up in the list of designs on the lab website. However, I was able to access the design by manually crafting the URL http://localhost:8000/new/lumina

I got a crash. when adjusting Waistband. To reproduce: Using the leftmost curated measurement set, slide Waistband Size down towards 0%. `points.snippetPanelBack` is unable to be created properly. `waistbandsize: 0.034`

In the Waistband part with Paperless enabled, when I slide Waist Lowering the dimension in the center jumps around from side to side. And, sometimes I see 2 dimensions, but other times I see only 1 dimension. The source code suggests there should be 2 dimensions. Are these dimensions measuring the things the are intended to measure? Or, were the wrong points accidentally used?

At the very top of the Leg part with Paperless enabled, when I slide Smoothing an upside-down dimension appears and disappears intermittently on the left.

woutervdub commented on designs/lumira/src/shape.mjs:
```diff
@@ -20,16 +20,17 @@ export const controlPoints = (p1, p2, p3, cpDistanceDivider) => {
+    return { cp1: p1.clone(), cp2: cp2 }
   }
+  if (p3 !== undefined && p2.sitsRoughlyOn(p3)) {
+    return { cp1: cp1.clone(), cp2: p3.clone() }
```
Good catch on the `clone()`

woutervdub commented:
> I got a crash. when adjusting Waistband. To reproduce: Using the leftmost curated measurement set, slide Waistband Size 
> down towards 0%. `points.snippetPanelBack` is unable to be created properly. `waistbandsize: 0.034`

Fixed. If a waistband is too small, a note is generated, and no part is drawn.

 
> In the Waistband part with Paperless enabled, when I slide Waist Lowering the dimension in the center jumps around from side to side. And, sometimes I see 2 dimensions, but other times I see only 1 dimension. The source code suggests there should be 2 dimensions. Are these dimensions measuring the things the are intended to measure? Or, were the wrong points accidentally used?

Fixed


> At the very top of the Leg part with Paperless enabled, when I slide Smoothing an upside-down dimension appears and disappears intermittently on the left.

There can be an additional dimension if the left or right most part of the leg sticks out from the top point on each side. This dimension was upside down. Corrected.

BenJamesBen commented:
I didn't look at the code changes, but I briefly tested the updated PR in the lab.

Even after running `yarn reconfigure`, Lumina didn't appear on the localhost lab web page. So, I assume that this means there is at least one more file change that will need to be added to the PR before it is ready for merge. (I'm no longer familiar with how this part of it works, ever since the update to v3. I'm not sure exactly what files need to be added/changed and included in the PR when adding a new design.)

Having the 2 designs, Lumina and Lumira, with very similar names has been causing confusion and making things a bit difficult when I try to work with either. 

Reducing the Waistband size no longer causes a crash. When it gets too small, the part no longer appears and a designer note is generated. However, if Waistband size is further reduced down to 0%, the designer note disappears as well.

woutervdub commented:
> I didn't look at the code changes, but I briefly tested the updated PR in the lab.
> 
> Even after running `yarn reconfigure`, Lumina didn't appear on the localhost lab web page. So, I assume that this means there is at least one more file change that will need to be added to the PR before it is ready for merge. 

That sits in `/config/software/designs.json`, and when multiple branches want to make changes, it becomes 'difficult'. Best to let @joostdecock take care of that when the PR is going to be merged.

> Having the 2 designs, Lumina and Lumira, with very similar names has been causing confusion and making things a bit difficult when I try to work with either.

Sorry. At the time I thought it was funny that two somewhat identical designs could have somewhat identical names. If people feel that this is an issue, I can fix that.
 
> Reducing the Waistband size no longer causes a crash. When it gets too small, the part no longer appears and a designer note is generated. However, if Waistband size is further reduced down to 0%, the designer note disappears as well.

Well, I would argue that a waistband of 0% is not a waistband at all.
